File: src/main/java/com/maximebertheau/emoji/Category.java
package com.maximebertheau.emoji;

/**
 * Created by Maxime Bertheau on 5/22/17.
 */
public enum Category {
    SYMBOLS,
    OBJECTS,
    NATURE,
    PEOPLE,
    FOODS,
    PLACES,
    ACTIVITY,
    FLAGS,
    ;

    public static Category parse(String input) {
        if(input == null) return null;

        input = input.toUpperCase();

        for (Category category : values()) {
            if (category.name().equals(input))
                return category;
        }

        return null;
    }
}

File: src/main/java/com/maximebertheau/emoji/Category.kt
package com.maximebertheau.emoji

/**
 * Created by Maxime Bertheau on 5/22/17.
 */
enum class Category(val dataName: String) {
    SMILEYS("Smileys & Emotion"),
    SYMBOLS("Symbols"),
    OBJECTS("Objects"),
    NATURE("Animals & Nature"),
    PEOPLE("People & Body"),
    FOODS("Food & Drink"),
    PLACES("Travel & Places"),
    ACTIVITY("Activities"),
    FLAGS("Flags"),
    SKIN_TONES("Skin Tones");

    companion object {
        @JvmStatic
        fun parse(name: String) = values().firstOrNull { it.dataName == name } ?: throw Error("Unknown category $name")
    }
}
